File: numbers/fibonacci.py
#!/usr/bin/env python

def fib(nth):
    if nth < 0:
        raise ValueError("Argument passed to fib must be non-negative integer");
    
    twoBehind = 0;
    oneBehind = 1;
    term = 1;
    
    if nth == 0:
        return twoBehind;
    
    if nth == 1:
        return oneBehind;
    
    index = 2;
    while index <= nth:
        term = oneBehind + twoBehind;
        twoBehind = oneBehind;
        oneBehind = term;
        index += 1;
    return term;

index = raw_input("Fibonacci index: ");
numInput = int(index);
listOfNums = [numInput];
counter = 0;
NUMS_TO_INCLUDE = 5;
while counter < NUMS_TO_INCLUDE:
    print fib(numInput + counter);
    counter += 1;

File: ruby/numbers/fibonacci.rb
#!/usr/bin/env ruby

#Let's try something new this time
#Let's read in our arguments
#through the command line

def main()
	if ARGV.length != 1
		puts "Usage: #{$0} index_of_fibonacci_number"
		exit(1)
	end

	begin
		input = Integer(ARGV[0])
		rescue ArgumentError => error
			puts "Input argument must be natural number"
			exit 1
	end

	puts "The Fibonacci number at position #{input} is #{fib(input)}"
end

def fib(n)
	if n == 0
		return 0
	end

	if n == 1
		return 1
	end

	behindBy1 = 1
	behindBy2 = 0

	index = 1
	while index < n
		term = behindBy1 + behindBy2
		index += 1
		behindBy2 = behindBy1
		behindBy1 = term
	end
	return term
end

main
